src/cargo/util/flock.rs

```diff
@@ -392,18 +393,20 @@ fn acquire(
     lock_try: &dyn Fn() -> io::Result<()>,
     lock_block: &dyn Fn() -> io::Result<()>,
 ) -> CargoResult<()> {
-    if cfg!(debug_assertions) {
-        // Force borrow to catch invalid borrows outside of contention situations
-        gctx.shell().verbosity();
-    }
+    let verbose = gctx.shell().verbosity() == Verbosity::Verbose;
     if try_acquire(path, lock_try)? {
         return Ok(());
     }
-    let msg = format!("waiting for file lock on {}", msg);
+    let path = path.display();
+    let msg = if verbose {
```

Comment:
Unsure if the discoverability of this behind `--verobse`. Personally I am fine with just printing the lock path unconditionally, though I think per [our contribution policy](https://doc.crates.io/contrib/process/working-on-cargo.html#before-hacking-on-cargo) maybe we want to discuss this in the original issue <https://github.com/rust-lang/cargo/issues/15094>?